test/results_test.exs

```diff
@@ -4,35 +4,41 @@ defmodule Wanda.ResultsTest do
 
   import Wanda.Factory
 
+  alias Wanda.Execution.Target
+
   alias Wanda.Results
 
   alias Wanda.Results.ExecutionResult
 
-  describe "Appending Execution Results to the history log" do
-    test "should correctly add an item to an empty log" do
+  describe "Creating an Execution" do
+    test "should correctly create a running execution in an empty system" do
       execution_id = UUID.uuid4()
       group_id = UUID.uuid4()
 
-      Results.create_execution_result!(
-        build(
-          :result,
-          execution_id: execution_id,
-          group_id: group_id,
-          result: :passing
-        )
-      )
+      [
+        %Target{
+          agent_id: agent_id,
+          checks: checks
+        }
+      ] = targets = build_list(1, :target)
+
+      Results.create_execution_result!(execution_id, group_id, targets)
 
       assert %ExecutionResult{
                execution_id: ^execution_id,
                group_id: ^group_id,
-               payload: %{
-                 "result" => "passing",
-                 "check_results" => [_ | _]
-               }
+               status: :running,
+               payload: %{},
+               targets: [
+                 %{
+                   agent_id: ^agent_id,
+                   checks: ^checks
+                 }
+               ]
              } = Repo.one!(ExecutionResult)
     end
 
-    test "should correctly add an item to a non empty log" do
+    test "should correctly create a running execution in a non empty system" do
```

Comment:
i think this tests are going to far aka testing that the ecto create functionality work. We can remove this test, and rename the one above "should create an execution in a starting running state"

Reply:
Good point, collapsed the two tests to just one https://github.com/trento-project/wanda/pull/60/commits/0587367f3124fd67218c92a5498440dcdf03c304